tests/gnrc_netif/main.c

```diff
@@ -450,6 +450,28 @@ static void test_ipv6_addr_best_src__other_subnet(void)
     TEST_ASSERT(!ipv6_addr_is_unspecified(out));
 }
 
+static void test_ipv6_addr_best_src__ula_src_dst(void)
+{
+    static const ipv6_addr_t ula_src = { .u8 = NETIF0_IPV6_ULA };
+    static const ipv6_addr_t ula_dst = { .u8 = { ULA1, ULA2, ULA3, ULA4,
+                                                 ULA5, ULA6, ULA7, ULA8,
+                                                 0, 0, 0, 0, 0, 0, 0, 1 } };
+    ipv6_addr_t *out = NULL;
+    int idx;
+
+    test_ipv6_addr_add__success();
+    TEST_ASSERT(0 <= (idx = gnrc_netif_ipv6_addr_add_internal(netifs[0], &ula_src, 64U,
```

Comment:
It might be good to also add the case with address (flags) set TENTATIVE and verify its not selected? Same when a "real" (non-ULA) global address is present. Also with and w/o default route set.

Reply:
Sounds like a good idea, however this seems to be a different topic than this PR (which I would like to merge before doing another to avoid merge conflicts) ;-).

Reply:
well the TENTATIVE case should be tested for ULAs too, bc that's what we also want to verify, right? Adding tests for real/global addresses could be another PR.